### DISCUSSION TITLE
TheXTech v1.3.5.3: Bugfix update with minor additions

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#236] [General]
This update fixes several bugs and crashes happen on previous 1.3.5.2 release, and also, introduces a small amount of brand new features.

**This release is a checkpoint prior to starting the big work on the 1.3.6 release actually.**

**Note:** To make use of TheXTech's full potential while designing new levels, the special [Moondust Devkit configuration package](https://github.com/Wohlstand/TheXTech-SDK) is recommended (the latest laboratory Moondust Devkit is required until the next stable devkit version 0.4.3.1 release).

## Changes for 1.3.5.3
- Fixed the vanilla crash: Attempting to freeze any Pokey segments when they have the "noiceball=0" option set will cause the out-of-range error with NPC-ID=-3 in a random moment. (https://github.com/Wohlstand/TheXTech/issues/220)
- Changed the user directory location for macOS-built games into the "~/TheXTech Games/<bundle name without .app suffix>/" to avoid conflict between different games ran on the same device.
- Changed the debug assets location on macOS into "~/TheXTech Games/Debug Assets/" to simplify the debug environment installation.
- Fixed BGO-65, Bowser III, and rail platforms compatibility with SMBX build version up to 30
- Fixed the false positive that enables the Yoshi Mode of music when a player actually doesn't ride any Yoshi
- Mario and Luigi will be only available at episodes built with SMBX older than build version 30 (the first version that introduces Peach, Toad, and Link)
- Fixed the invalid vertical offset while drawing a playable character sprite
- Resolved the old jittering problem when drawing a playable character while moving the camera (thanks to @ds-sloth for the fix)
- Fixed the minor graphical distortion of a playable character that appears while flying up and down with a shoe.
- The last played episode will be reminded and scrolled at the episode selection menu
- Improved episodes menu control: added mouse wheel, page-up/page-down/home/end/delete, and left/right/alt-run/alt-jump/drop on controllers to scroll episodes lists
- Fixed the vanilla bug: episodes list menu gets reset every attempt to go back from the character or the game save slot select menu
- Fixed the soft-lock in an attempt to erase the gamesave using the two-player game menu
- Order of episodes lists now alphabetical
- Added the "compat.ini" option to require players to be on the ground to enter warps
- On-screen elements such as HUD and other printing debug information will not shake with in-game happenings
- Added screen fading effects at the world map and levels
- Added support for different transition effects per every warp entry
- Fixed the vanilla bug: The deadlock in the form of a looped message box showing will happen in the condition: the first NPC activates another NPC using recursive activation of all touching NPCs. And when the second NPC has an on-activation event trigger that hides its layer and shows a message box
- Fixed the unexpected Wart's death sound spam

## Known issues
* Sometimes horizontal piranha plants would cause a small missing of a pixels column
* Affected by [DrPepper Problem](https://wohlsoft.ru/pgewiki/DrPepper_Problem): The strong lag on slow devices when the level has any moving layers that move horizontally (by X-axis), mainly because the blocks search indexing gets completely disabled when the layer gets moved by X-axis. Todo: implement the QuadTree for all level objects and make sure they properly getting updated and found. (Issue #33, got been resolved at the `devel` branch by @ds-sloth)
* Can't type any cheats on a mobile device without a USB-OTG keyboard yeh. Some levels (for example, the "Superb Demo 128" at "Prelude To The Stupid") do require the use of cheats for being passed, otherwise, these levels become unbeatable without a USB-OTG keyboard plugged. Todo: implement the calling of the software keyboard to allow running cheats on mobile devices. (Issue #61)

## Showcases
![Scr_2021-12-19_09-33-12](https://user-images.githubusercontent.com/6751442/147757926-5c80509a-d500-4901-859d-54e068f019a7.gif)

![Scr_2021-12-19_09-34-56](https://user-images.githubusercontent.com/6751442/147757943-97194c6b-8d78-449e-8f57-bab4a1b0a81d.gif)

![Scr_2021-12-29_07-02-23](https://user-images.githubusercontent.com/6751442/147758004-3245c173-6ad6-4813-950a-dbacad385321.gif)

![Scr_2021-12-29_08-10-26](https://user-images.githubusercontent.com/6751442/147758032-99a7e96c-ebda-4488-b8d4-3e9d19044f7b.gif)

<hr /><em>This discussion was created from the release <a href='https://github.com/Wohlstand/TheXTech/releases/tag/v1.3.5.3'> TheXTech v1.3.5.3: Bugfix update with minor additions</a>.</em>

**Comment by LoveBodhi** [comment]
I see, original SMBX1.3 asset should be add the settings for SMBX 1.3 compatible mode ("smbx13")

[thextech [SMBX13 Config File].zip](https://github.com/Wohlstand/TheXTech/files/7797563/thextech.SMBX13.Config.File.zip)

**Comment by Wohlstand** [comment]
Hi there!

Just now I released a hotfix to resolve the problem of autostart events that may cause a black screen when message boxes were used.

https://github.com/Wohlstand/TheXTech/releases/tag/v1.3.5.3-1

**Comment by LoveBodhi** [comment]
Now toggle Status Bar is available.
